Tests fix after InViewportMixin was removed

sergeykers opened the pull request:
1. my-component.js: transformed to class; switched from using InViewportMixin to InViewportService; removed unused properties
2. app.css: top: 0 added to fix failed test issue when using browser window
3. my-component.hbs: template created
4. in-viewport-test.js: InViewportMixin test removed
5. viewport-config.js, my-component-test.js, README.md: removed unused properties (were used in deleted mixin)
6. integration-test.js: added waitFor() for time to rendering changes
7. index.hbs: removed test of a deleted property
8. application.hbs: two links on one page made one

Commit:
#Comment Fixed several tests using deleted InViewportMixin
1. my-component.js: transformed to class; switched from using InViewportMixin to InViewportService
2. app.css: top: 0 added to fix failed test issue when using browser window
3. my-component.hbs: template created
4. in-viewport-test.js: InViewportMixin test removed

Commit:
#Comment Fixed or removed several tests, cleared unused properties
1. viewport-config.js, my-component.js, my-component-test.js, README.md: removed unused properties (used in deleted mixin)
2. integration-test.js: added short delays for time to rendering changes
3. index.hbs: removed test of a deleter property

Commit:
#Comment Fixed some minor bugs
1. integration-test.js: removed redundant options and comments
2. my-component.js: willDestroy() hook added
3. application.hbs: two links on one page made one

andrei-sviazhin commented on tests/unit/mixins/in-viewport-test.js:
```diff
@@ -1,6 +1,3 @@
-import InViewportMixin from 'ember-in-viewport/mixins/in-viewport';
 import { module, test } from 'qunit';
 import { setupTest } from 'ember-qunit';
 import RafPool from 'raf-pool';
```
@SergeAstapov Name of this test now is `Unit | Mixin | in viewport`, but we've remove `InViewportMixin` from here  and it actually became a test for RafPool package only. Should we keep this test at all?

SergeAstapov replied:
@sergeykers let's keep the test but rename the file to `tests/unit/raf-pool-test.js` and change the line

```js
module('Unit | Mixin | in viewport', function (hooks) {
```

to

```js
module('Unit | Raf Pool', function (hooks) {
```

sergeykers commented:
@SergeAstapov, do you know, why https://github.com/SergeAstapov/ember-in-viewport/blob/master/addon/-private/raf-admin.js is called "mixin" in the comment at the top? Should it be called that way?

SergeAstapov commented:
> @SergeAstapov, do you know, why https://github.com/SergeAstapov/ember-in-viewport/blob/master/addon/-private/raf-admin.js is called "mixin" in the comment at the top? Should it be called that way?

I think just some remnants of previous changes in the codebase and potentially some copy-paste so we should be able to safely delete replace word "mixin" with "class"

SergeAstapov commented:
@sergeykers there are changes in `package-lock.json` but no changes in `package.json` - let's revert that file to avoid issues.

As it has `"lockfileVersion": 1`, you need to make sure you use npm < 7

SergeAstapov commented on tests/dummy/app/components/my-component.js:
```diff
@@ -44,12 +44,32 @@ export default Component.extend(InViewportMixin, {
+    onExit(this.didExitViewport.bind(this));
+  }
+
+  constructor() {
```
@sergeykers lets remove this `constructor` as it does nothing

SergeAstapov commented on tests/dummy/app/templates/infinity-right-left.hbs:
```diff
@@ -16,7 +16,7 @@
       @infinityLoad={{this.infinityLoad}}
     >
-      &npbs;
+      &nbsp;
```
nice catch!

SergeAstapov commented on tests/unit/mixins/in-viewport-test.js:
```diff
@@ -1,6 +1,3 @@
-import InViewportMixin from 'ember-in-viewport/mixins/in-viewport';
 import { module, test } from 'qunit';
 import { setupTest } from 'ember-qunit';
 import RafPool from 'raf-pool';
```
@sergeykers I believe this test file needs to be renamed as well as test module because there is no more mixin involved

Commit:
#Comment Fixes after the review results
1. package-lock.json changes has been reverted
2. my-component.js: redundant constructor() method removed
3. in-viewport-test.js renamed to raf-pool-test.js and moved from 'mixins' to 'dependencies'

Commit:
#Comment Fixes after the review results
1. raf-pool-test.js moved to tests/unit.
2. Test module renamed.
